client/src/partials/templates/modals/confirmDeletion.modal.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+.controller('ConfirmDeletionModalController', ConfirmDeletionModalController);
+
+// dependencies injection
+ConfirmDeletionModalController.$inject = ['$uibModalInstance', '$translate', 'data'];
+
+/**
+ * Confirm Deletion Controller
+ * This controller is responsible for check a match text given to continue
+ * the deletion process
+ */
+function ConfirmDeletionModalController (Instance, $translate, Data) {
+  var vm = this;
+
+  // Global objects
+  vm.pattern = Data.pattern;
+  vm.elementName = Data.elementName;
```

Comment:
In order to avoid programmer error, please put a check to make sure that both the `pattern` and the `elementName` are defined.  Something like this:

```js
// make sure the modal isn't accidentally called with empty values
if (!Data.pattern || !Data.elementName) {
  throw new Error(
    'ConfirmDeletionModal requires both a pattern and an element name to be defined, but only received:' +
    'data.pattern = ' + Data.pattern + ' ' +
    'data.elementName = ' + Data.elementName
  );
}
```
It would help prevent against silly errors that programmers make in the future.